uberfire-nio2-backport/uberfire-nio2-model/src/main/java/org/uberfire/java/nio/base/dotfiles/DotFileUtils.java

```
@@ -36,11 +38,46 @@ private DotFileUtils() {
 
     }
 
-    public static boolean buildDotFile(final Path path,
-                                       final OutputStream out,
-                                       final FileAttribute<?>... attrs) {
+    public static boolean buildDotFile( final Path path, final FileAttribute<?>... attrs ) {
+        Path dotPath = dot( path );
+        OutputStream out = path.getFileSystem().provider().newOutputStream( dotPath );
         boolean hasContent = false;
-        if (attrs != null && attrs.length > 0) {
+        try {
+            hasContent = buildDotFile( path, out, attrs);
+        }
+        finally {
+            try {
+                out.close();
+            } catch (Exception e) {
+            }
+        }
+        return hasContent;
+    }
+
+    public static boolean buildDotFile( final Path path, File dotFile, final FileAttribute<?>... attrs ) {
+        OutputStream out = null;
+        boolean hasContent = false;
+        try {
+            out = new FileOutputStream( dotFile );
+            hasContent = buildDotFile( path, out, attrs);
+        }
+        catch (Exception e) {
+        }
+        finally {
+            try {
+                out.close();
+            } catch (Exception e) {
+            }
```

Empty catch block, nested try catch